sim/team-validator.js

```diff
@@ -641,7 +640,6 @@ class Validator {
 				if (fastReturn) return true;
 				problems.push(`${name}'s gender must be ${eventData.gender}${etc}.`);
 			}
-			if (!fastReturn) set.gender = eventData.gender;
```

Comment:
The gender mismatch is already checked here, so if the event passes validation and has a gender then the set either did not specify a gender (in which case we want to set it) or already has the correct gender.